mlflow/xgboost.py

```diff
@@ -0,0 +1,174 @@
+"""
+The ``mlflow.xgboost`` module provides an API for logging and loading XGBoost models.
+This module exports XGBoost models with the following flavors:
+
+XGBoost (native) format
+    This is the main flavor that can be loaded back into XGBoost.
+:py:mod:`mlflow.pyfunc`
+    Produced for use by generic pyfunc-based deployment tools and batch inference.
+"""
+
+from __future__ import absolute_import
+
+import os
+import yaml
+
+import mlflow
+from mlflow import pyfunc
+from mlflow.models import Model
+from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+
+FLAVOR_NAME = "xgboost"
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`.
+    """
+    import xgboost as xgb
+
+    return _mlflow_conda_env(
+        additional_conda_deps=None,
+        additional_pip_deps=[
+            "xgboost=={}".format(xgb.__version__),
+        ],
+        additional_conda_channels=None)
+
+
+def save_model(xgb_model, path, conda_env=None, mlflow_model=Model()):
+    """
+    Save an XGBoost model to a path on the local file system.
+
+    :param xgb_model: XGBoost model to be saved.
+    :param path: Local path where the model is to be saved.
+    :param conda_env: Either a dictionary representation of a Conda environment or the path to a
+                      Conda environment yaml file. If provided, this describes the environment
+                      this model should be run in. At minimum, it should specify the dependencies
+                      contained in :func:`get_default_conda_env()`. If ``None``, the default
+                      :func:`get_default_conda_env()` environment is added to the model.
+                      The following is an *example* dictionary representation of a Conda
+                      environment::
+
+                        {
+                            'name': 'mlflow-env',
+                            'channels': ['defaults'],
+                            'dependencies': [
+                                'python=3.7.0',
+                                'pip': [
+                                    'xgboost==0.90'
+                                ]
+                            ]
+                        }
+
+    :param mlflow_model: :py:mod:`mlflow.models.Model` this flavor is being added to.
+    """
+    import xgboost as xgb
+
+    path = os.path.abspath(path)
+    if os.path.exists(path):
+        raise Exception("Path '{}' already exists".format(path))
```

Comment:
```suggestion
        raise MlflowException("Path '{}' already exists".format(path), DIRECTORY_NOT_EMPTY)
```

The suggestion is to raise the more specific MlflowException so users can catch that if they want (note that they can still catch a broader `Exception`)

Reply:
will fix this